api/src/main/java/uk/gov/dvsa/mot/app/TradeServiceRequestHandler.java

```diff
@@ -108,9 +109,16 @@ public void setTradeReadService(TradeReadService tradeReadService) {
                 List<Vehicle> vehicles = tradeReadService.getVehiclesByRegistration(registration);
 
                 if (CollectionUtils.isNullOrEmpty(vehicles)) {
-                    logger.debug("getTradeMotTests for registration = " + request.getQueryParams().getNumber() + " found 0");
-                    throw new InvalidResourceException("No MOT Tests found with vehicle registration : " + registration,
-                            context.getAwsRequestId());
+
```

Comment:
@rstepnowski why adding logic to controller? I think the service method should be extended in similar way to uk.gov.dvsa.mot.trade.read.core.TradeReadServiceDatabase#getLatestMotTestByRegistration